tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,289 @@
+
+        while(timeout_ms < 0 ? 1 : test_clock() <= tmout) {
+                test_consumer_poll_once(c->rk, c->mv, 1000);
+                if (*target > start)
+                        return 1;
+              }
+
+        return 0;
+}
+
+static void rebalance_cb (rd_kafka_t *rk,
+                          rd_kafka_resp_err_t err,
+                          rd_kafka_topic_partition_list_t *parts,
+                          void *opaque) {
+        int i;
+        _consumer_t *c = opaque;
+
+        switch (err)
+        {
+        case RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS:
+                c->partition_cnt = parts->cnt;
+                c->assigned_at = test_clock();
+
+                rd_kafka_assign(rk, parts);
+
+                TEST_SAY("%s Assignment (%d partition(s)): ",
+                         rd_kafka_name(rk), parts->cnt);
+                for (i = 0 ; i < parts->cnt ; i++)
+                        TEST_SAY0("%s%s[%"PRId32"]",
+                                  i == 0 ? "" : ", ",
+                                  parts->elems[i].topic,
+                                  parts->elems[i].partition);
+                TEST_SAY0("\n");
+
+                break;
+
+        case RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS:
+                TEST_ASSERT(++c->rebalance_cnt == c->max_rebalance_cnt,
+                    "%s rebalanced %d times, max was %d",
+                    rd_kafka_name(rk),
+                    c->rebalance_cnt, c->max_rebalance_cnt);
+
+                TEST_SAY("%s revoked %d partitions\n", rd_kafka_name(c->rk), parts->cnt);
+                c->revoked_at = test_clock();
+                rd_kafka_assign(rk, NULL);
+                break;
+
+        default:
+                TEST_FAIL("rebalance failed: %s", rd_kafka_err2str(err));
+                break;
+        }
+
+}
+
+int main_0102_static_group_rebalance (int argc, char **argv) {
+        rd_kafka_conf_t *conf;
+        test_msgver_t mv;
+        int64_t rebalance_start;
+
+        _consumer_t c[_CONSUMER_CNT] = RD_ZERO_INIT;
+        const int msgcnt = 100;
+        uint64_t testid  = test_id_generate();
+        const char *topic = test_mk_topic_name("0102_static_group_rebalance",
+                                               1);
+        char *topics = rd_strdup(tsprintf("^%s.*", topic));
+
+
+        test_msgver_init(&mv, testid);
+        c[0].mv = &mv;
+        c[1].mv = &mv;
+
+        test_create_topic(NULL, topic, 3, 1);
+        test_produce_msgs_easy(topic, testid, RD_KAFKA_PARTITION_UA, msgcnt);
+
+        test_conf_init(&conf, NULL, 60);
+        test_conf_set(conf, "session.timeout.ms", "5000");
+        test_conf_set(conf, "max.poll.interval.ms", "10001");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "topic.metadata.refresh.interval.ms", "500");
+        test_conf_set(conf, "enable.partition.eof", "true");
+        test_conf_set(conf, "isolation.level", "read_uncommitted");
```

Comment:
What is this for? You're not producing transactional messages to the topic, right?